staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -320,6 +327,40 @@ func (o *WaitOptions) RunWait() error {
 	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), o.Timeout)
 	defer cancel()
 
+	isForDelete := strings.ToLower(o.ForCondition) == "delete"
+	if o.WaitForCreation && o.Timeout == 0 {
```

Comment:
suggest collapsing this into the if block below and validating all the things that cannot be used with `o.WaitForCreation`